pkg/R/utils.R

```diff
@@ -164,11 +176,14 @@ reserialize <- function(rdd) {
   if (!inherits(rdd, "RDD")) {
     stop("Argument 'rdd' is not an RDD type.")
   }
-  if (rdd@env$serialized) {
-    return(rdd)
-  } else {
+  if (rdd@env$serializedMode != "byte") {
+    if (rdd@env$serializedMode == "row") {
+      rdd@env$serialzedMode == "byte"
```

Comment:
I think I've gotten to the bottom of this. Here is my take, @cafreeman @concretevitamin, please correct me if I am wrong.

The problem is in initializing `PipelinedRDD` where we set `serializedMode` - We need the following diff there 
```
-   # This tracks the serialization mode for jrdd_val
-  .Object@env$serializedMode <- prev@env$serializedMode
+  if (!is.null(jrdd_val)) {
+     # This tracks the serialization mode for jrdd_val
+    .Object@env$serializedMode <- prev@env$serializedMode
+  }
```
as we shouldn't set `serializedMode` if `jrdd_val` is NULL (basically the invariant I see is that in PipelinedRDD serializedMode tracks jrdd_val). 

This results in the need for one more change: 
```
-    .Object@env$prev_serializedMode <- .Object@env$serializedMode
+    .Object@env$prev_serializedMode <- prev@env$serializedMode  
```
where we'll set `prev_serializedMode` using `prev@env$serializedMode`

After this we don't need the change in `reserialize` -- The one thing to note is that we shouldn't always assume rdd@env$serialized exists in unit tests or things like that. Basically we don't know the serialization mode for some RDDs if we don't have a JRDD associated with them.

Let me know if this makes sense !